### ISSUE TITLE
php-sha3 cannot run on macOS 10.12.3/PHP 5.6.28/Apache 2.4.23

### ISSUE THREAD

**Issue by windcloud-409** [strawbrary/php-sha3#14]
I follow the README to install and when I try to execute `echo sha3('');` in php (console), I get error:
`[1]    11937 bus error  php testsha3.php`

But in phpinfo I found that sha3 is already enabled.

There're the log in console:
```
 ~/Documents/php-sha3 phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226


 ~/Documents/php-sha3 ./configure --enable-sha3
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-apple-darwin16.4.0
checking host system type... x86_64-apple-darwin16.4.0
checking target system type... x86_64-apple-darwin16.4.0
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib
checking for PHP extension directory... /usr/lib/php/extensions/no-debug-non-zts-20131226
checking for PHP installed headers prefix... /usr/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... no
checking for nawk... no
checking for awk... awk
checking if awk is broken... no
checking Whether to enable SHA3 support... yes, shared
checking for ld used by cc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /usr/bin/nm -B
checking whether ln -s works... yes
checking how to recognize dependent libraries... pass_all
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking the maximum length of command line arguments... 196608
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking if cc supports -fno-rtti -fno-exceptions... yes
checking for cc option to produce PIC... -fno-common
checking if cc PIC flag -fno-common works... yes
checking if cc static flag -static works... no
checking if cc supports -c -o file.o... yes
checking whether the cc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.4.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no

creating libtool
appending configuration tag "CXX" to libtool
configure: creating ./config.status
config.status: creating config.h


 ~/Documents/php-sha3 make
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/php_sha3.c -o php_sha3.lo
mkdir .libs
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/php_sha3.c  -fno-common -DPIC -o .libs/php_sha3.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/KeccakHash.c -o KeccakHash.lo
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/KeccakHash.c  -fno-common -DPIC -o .libs/KeccakHash.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/KeccakSponge.c -o KeccakSponge.lo
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/KeccakSponge.c  -fno-common -DPIC -o .libs/KeccakSponge.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/KeccakF-1600-reference.c -o KeccakF-1600-reference.lo
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/KeccakF-1600-reference.c  -fno-common -DPIC -o .libs/KeccakF-1600-reference.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/SnP-FBWL-default.c -o SnP-FBWL-default.lo
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/SnP-FBWL-default.c  -fno-common -DPIC -o .libs/SnP-FBWL-default.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=compile cc  -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /Users/usera/Documents/php-sha3/displayIntermediateValues.c -o displayIntermediateValues.lo
 cc -I. -I/Users/usera/Documents/php-sha3 -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /Users/usera/Documents/php-sha3/displayIntermediateValues.c  -fno-common -DPIC -o .libs/displayIntermediateValues.o
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=link cc -DPHP_ATOM_INC -I/Users/usera/Documents/php-sha3/include -I/Users/usera/Documents/php-sha3/main -I/Users/usera/Documents/php-sha3 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -o sha3.la -export-dynamic -avoid-version -prefer-pic -module -rpath /Users/usera/Documents/php-sha3/modules  php_sha3.lo KeccakHash.lo KeccakSponge.lo KeccakF-1600-reference.lo SnP-FBWL-default.lo displayIntermediateValues.lo
cc ${wl}-flat_namespace ${wl}-undefined ${wl}suppress -o .libs/sha3.so -bundle  .libs/php_sha3.o .libs/KeccakHash.o .libs/KeccakSponge.o .libs/KeccakF-1600-reference.o .libs/SnP-FBWL-default.o .libs/displayIntermediateValues.o
dsymutil .libs/sha3.so || :
creating sha3.la
(cd .libs && rm -f sha3.la && ln -s ../sha3.la sha3.la)
/bin/sh /Users/usera/Documents/php-sha3/libtool --mode=install cp ./sha3.la /Users/usera/Documents/php-sha3/modules
cp ./.libs/sha3.so /Users/usera/Documents/php-sha3/modules/sha3.so
cp ./.libs/sha3.lai /Users/usera/Documents/php-sha3/modules/sha3.la
----------------------------------------------------------------------
Libraries have been installed in:
   /Users/usera/Documents/php-sha3/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `DYLD_LIBRARY_PATH' environment variable
     during execution

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.


 ~/Documents/php-sha3 sudo make install
Installing shared extensions:     /usr/lib/php/extensions/no-debug-non-zts-20131226/
```

How to fix this error?

**Comment by joshuascottpaul** [comment]
same issue:
$ phpize
Configuring for:
PHP Api Version:         20131106
Zend Module Api No:      20131226
Zend Extension Api No:   220131226

Installing shared extensions:     /usr/lib/php/extensions/no-debug-non-zts-20131226/

sha3
sha3 support => enabled

$ php -a
Interactive shell

php > echo sha3('');
Bus error: 10

**Comment by tml** [comment]
PR #16 fixes this.

**Comment by strawbrary** [comment]
Fixed by #16
